src/swell/test/code_tests/test_generate_observing_system.py

```diff
@@ -0,0 +1,90 @@
+import os
+import unittest
+import subprocess
+from datetime import datetime as dt
+from swell.utilities.logger import Logger
+from swell.utilities.get_channels import get_channels
+from swell.utilities.observing_system_records import ObservingSystemRecords
+
+
+def setup_geos_mksi(branch):
+    url = "https://github.com/GEOS-ESM/GEOS_mksi.git"
+    # Clone repo if not already cloned
+    if not os.path.exists("GEOS_mksi"):
+        git_clone_cmd = ["git", "clone", "-b", branch, url, "GEOS_mksi"]
+        subprocess.run(git_clone_cmd, stderr=subprocess.DEVNULL)
+    else:
+        cwd = os.getcwd()
+        os.chdir("GEOS_mksi")
+        git_checkout_cmd = ['git', 'checkout', branch]
+        subprocess.run(git_checkout_cmd, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+        os.chdir(cwd)
```

Comment:
`os.chdir` is a bit fragile and also harder to reason about (you have to look back to the most recent call to remember exactly what directory you're in).

A more reliable approach is to tell `subprocess.run` exactly where the command should be running.

```suggestion
        git_checkout_cmd = ['git', 'checkout', branch]
        subprocess.run(git_checkout_cmd, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, cwd="./GEOS_mksi")
```

Reply:
Cool! Might also change `chdir` usage in `jedi_bundle` as well